utils/azure/vmsizes/vmsizes.py

```diff
@@ -41,19 +91,26 @@ def get_vmsizes(subscription_id, location, token):
         if "value" in dict_:
             data = dict_["value"]
             for item in data:
-                key = item["name"]
-                value = {
-                    'vcpus': item['vCPUs'],
-                    'ram': int(item['memoryGB'] * 1000),
-                    'gpus': item['gpus']
-                }
-                if key in output and value != output[key]:
-                    print(f"WARNING: {key} has conflicting values: {output[key]} and {value}")
-                else:
-                    output[key] = value
+                if item["resourceType"] == "virtualMachines":
+                    key = item["name"]
+                    value = {
+                        "vcpus": int(extract_value("vCPUs", item["capabilities"])),
+                        "ram": int(
+                            float(extract_value("MemoryGB", item["capabilities"]))
+                            * 1000
+                        ),
+                        "gpus": int(extract_value("GPUs", item["capabilities"]))
+                        if extract_value("gpus", item["capabilities"]) is not None
+                        else 0,
+                    }
```

Comment:
The value extraction can be simplified by defining a dictionary first:
```python
                    caps = { el["name"]:el["value"] for el in item["capabilities"] }
                    value = {
                        "vcpus": int(caps.get("vCPUsAvailable", caps["vCPUs"])),
                        "ram":   int(float(caps.get("MemoryGB"))* 1000),
                        "gpus":  int(caps.get("GPUs", 0))
                    }
```

This also fixes the case where for some VMs, `vCPUs != vCPUsAvailable`.